h/views/activity.py

```diff
@@ -87,7 +86,6 @@ def user_link(userid):
 
 @view_defaults(route_name='group_read',
                renderer='h:templates/activity/search.html.jinja2',
-               effective_principals=security.Authenticated,
```

Comment:
I wonder why this was here in the first place? It doesn't seem to achieve anything other than breaking the public group's page

Reply:
I tested visiting a private group's page when logged out, when logged in but not a member of the group, and when logged in and a member of the group - all seems to work as expected. So it looks like it's fine to remove this.

Reply:
I'm assuming this was to avoid a conflict with the "unauthenticated read" view in `h.views.groups`. Otherwise Pyramid would presumably complain about having multiple views matching a given request.

Reply:
Yes, that's right. Pyramid was deliberately configured so that _this_ view would be selected when you were logged in, but unauthenticated read would be selected when you were logged out.